opendistro-for-elasticsearch/hooks/run

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+exec 2>&1
+
+# Don't use the default password, ok?
+if [ "{{cfg.securityconfig.internal_users.admin.hash}}" = '$2a$12$VcCDgh2NDk07JGN0rjGbM.Ad41qVR/YFJcgHp0UGns5JDymv..TOG' ]; then
+    echo "=================================================="
+    echo "WARNING: UNSAFE DEFAULT PASSWORD FOR ADMIN USER!!!"
+    echo "WARNING: USE FOR TESTING PURPOSES ONLY!!!"
+    echo "WARNING: UNSAFE DEFAULT PASSWORD FOR ADMIN USER!!!"
+    echo "=================================================="
+fi
+
+free_mem="$(free -m | grep Mem | awk '{print $2}')"
+computed_mem="$(( $free_mem / 2 ))"
+half_mem="$(echo $computed_mem)"
+if [ $half_mem -lt 26624 ]; then
+    heapsize=$half_mem; heapsize+=m;
+else
+    heapsize=26624m;
+fi
```

Comment:
I wonder if we should auto-calculate this by default but allow the user to explicitly set this via config if they need something different.